File: medium/1653_min_deletions_to_make_string_balanced/main.cpp
/*
You are given a string s consisting only of characters 'a' and 'b'​​​​.

You can delete any number of characters in s to make s balanced. s is balanced if there is no pair of indices (i,j) such that i < j and s[i] = 'b' and s[j]= 'a'.

Return the minimum number of deletions needed to make s balanced.
*/

#include <string>

using namespace std;

class Solution
{
public:
    int minimumDeletions(string s)
    {
        int numBLeft = 0;
        int minDel = 0;

        for (auto c : s)
        {
            if (c == 'b')
            {
                numBLeft++;
            }
            else
            {
                minDel = min(minDel + 1, numBLeft);
            }
        }

        return minDel;
    }
};

File: medium/1653_min_deletions_to_make_string_balanced/main.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        min_del = 0
        b_left = 0

        for k in range(len(s)):
            if s[k] == 'b':
                b_left +=1
            else:
                min_del = min(min_del + 1, b_left)

        return min_del